### ISSUE TITLE
ExceptionMessages.ToEnvironmentSpecificResult() should ALWAYS return object of type ExceptionMessages

### ISSUE THREAD

**Issue by Skyaz** [anavarro9731/serviceapi-infrastructure#4]
https://github.com/anavarro9731/serviceapi-infrastructure/blob/master/ServiceApi.Infrastructure/Models/ExceptionMessages.cs#L29

Because we return an anonymous type in non-developement mode, this will cause exception not to be thrown or incorrect HTTP status code due to type-checking failure.

```
// HTTP example...

var result = await pipeline.Execute(apiMessage);
// PROD/UAT returns new { ErrorMessage }

// Not an Exception
if (result is Exception)
    result = ExceptionMessages.Create((Exception)result);

// Not an ExceptionMessages
if (result is ExceptionMessages)
    return BadRequest(result);

// WRONG!!! - 200 returned to client
return Ok(result);
```

```
// NServiceBus example...

var result = await messagePipeline.Execute(command).ConfigureAwait(false);
// PROD/UAT returns new { ErrorMessage }

// Not an ExceptionMessages
if (result is ExceptionMessages)
    result = new Exception(((ExceptionMessages)result).DetailedErrorMessage ?? ((ExceptionMessages)result).ErrorMessage);

// Not an Exception
if (result is Exception)
    throw result as Exception;

// WRONG!!! - No exception thrown, no NSB TX rollback
```

**Comment by anavarro9731** [comment]
Create a base class to ExceptionMessages?

**Comment by Skyaz** [comment]
Resolved in [commit 41085c2](https://github.com/anavarro9731/serviceapi-infrastructure/commit/41085c2432efe37428082cc429d728be4cd668a7)
